impl-tinkerpop-parent/impl/src/main/java/org/hawkular/inventory/impl/tinkerpop/TinkerpopBackend.java

```diff
@@ -0,0 +1,728 @@
+import org.hawkular.inventory.api.model.ElementBlueprintVisitor;
+import org.hawkular.inventory.api.model.ElementUpdateVisitor;
+import org.hawkular.inventory.api.model.ElementVisitor;
+import org.hawkular.inventory.api.model.Entity;
+import org.hawkular.inventory.api.model.Environment;
+import org.hawkular.inventory.api.model.Feed;
+import org.hawkular.inventory.api.model.Metric;
+import org.hawkular.inventory.api.model.MetricType;
+import org.hawkular.inventory.api.model.MetricUnit;
+import org.hawkular.inventory.api.model.Relationship;
+import org.hawkular.inventory.api.model.Resource;
+import org.hawkular.inventory.api.model.ResourceType;
+import org.hawkular.inventory.api.model.Tenant;
+import org.hawkular.inventory.api.paging.Page;
+import org.hawkular.inventory.api.paging.Pager;
+import org.hawkular.inventory.base.Query;
+import org.hawkular.inventory.base.spi.CanonicalPath;
+import org.hawkular.inventory.base.spi.ElementNotFoundException;
+import org.hawkular.inventory.base.spi.InventoryBackend;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.StreamSupport;
+
+import static java.util.stream.Collectors.toSet;
+import static org.hawkular.inventory.api.Relationships.Direction.incoming;
+import static org.hawkular.inventory.api.Relationships.Direction.outgoing;
+import static org.hawkular.inventory.api.Relationships.WellKnown.contains;
+import static org.hawkular.inventory.impl.tinkerpop.Constants.Type.environment;
+import static org.hawkular.inventory.impl.tinkerpop.Constants.Type.feed;
+import static org.hawkular.inventory.impl.tinkerpop.Constants.Type.metric;
+import static org.hawkular.inventory.impl.tinkerpop.Constants.Type.metricType;
+import static org.hawkular.inventory.impl.tinkerpop.Constants.Type.relationship;
+import static org.hawkular.inventory.impl.tinkerpop.Constants.Type.resource;
+import static org.hawkular.inventory.impl.tinkerpop.Constants.Type.resourceType;
+import static org.hawkular.inventory.impl.tinkerpop.Constants.Type.tenant;
+
+/**
+ * @author Lukas Krejci
+ * @since 0.1.0
+ */
+final class TinkerpopBackend implements InventoryBackend<Element> {
+    private final InventoryContext context;
+
+    public TinkerpopBackend(InventoryContext context) {
+        this.context = context;
+    }
+
+    @Override
+    public Element find(CanonicalPath element) throws ElementNotFoundException {
+        Iterator<Element> it = navigate(element);
+        if (!it.hasNext()) {
+            throw new ElementNotFoundException();
+        } else {
+            return it.next();
+        }
+    }
+
+    @Override
+    public Page<Element> query(Query query, Pager pager) {
+        HawkularPipeline<?, ? extends Element> q;
+        if (query.getFragments()[0].getFilter() instanceof RelationFilter) {
```

Comment:
It is the same thing as `JumpInOutFilter`. It is used in the base impl, afair. The backend just translates the Query to its native form - here a gremlin pipeline and executes it. There's still the `FilterApplicator` which mostly remained the same where you can see how it applies the `SwitchElementType` (which as you correctly found out is just a renamed `JumpInOutFilter`).